### PR TITLE
CIS-CAT wodle for Linux

### PR DESCRIPTION
**Pull request by chemamartinez**
It has been included a CIS-CAT module to schedule CIS-CAT scans in Linux agents.

For running this module, a section like the following one has to be included in `ossec.conf`:

```
  <wodle name="cis-cat">
    <disabled>no</disabled>
    <timeout>1800</timeout>
    <interval>1d</interval>
    <scan-on-start>yes</scan-on-start>

    <java_path>/usr/bin</java_path>
    <ciscat_path>/root/cis-cat-full</ciscat_path>

    <content type="xccdf" path="benchmarks/CIS_Ubuntu_Linux_16.04_LTS_Benchmark_v1.0.0-xccdf.xml">
      <profile>xccdf_org.cisecurity.benchmarks_profile_Level_2_-_Workstation</profile>
    </content>
  </wodle>
```

You can specify the fields described below:
 

- `Disabled, timeout, interval, scan-on-start` as performance options.
- `java_path, ciscat_path` to set where are located the necessary tools for running the scans.
- `content` section for specifying what benchmark is going to be assessed.

When a scan has finished, alerts will be generated including the scan results.

### CODE REVIEW

File: etc/rules/0505-ciscat_rules.xml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    <description>CIS-CAT events.</description>
+  </rule>
+
+  <rule id="87402" level="3">
+      <if_sid>87401</if_sid>
+      <field name="type">^scan_start$</field>
+      <description>CIS-CAT: assessment information for scan $(scan_id)</description>
+  </rule>
+
+  <rule id="87403" level="5">
```
Comment (vikman90) [inline etc/rules/0505-ciscat_rules.xml]: The rule level for correct results (_pass_, _notchecked_, _notselected_) should be `0` and `1` for _informational_. Please make these rules behave like https://github.com/wazuh/wazuh/blob/master/etc/rules/0385-oscap_rules.xml.

File: src/wazuh_modules/wm_ciscat.c [hunk 2]
```diff
@@ -0,0 +1,530 @@
...
+
+        if ((time_t)ciscat->interval >= time_sleep) {
+            time_sleep = ciscat->interval - time_sleep;
+            ciscat->state.next_time = ciscat->interval + time_start;
+        } else {
+            mterror(WM_CISCAT_LOGTAG, "Interval overtaken.");
+            time_sleep = ciscat->state.next_time = 0;
+        }
+
+        if (wm_state_io(&WM_CISCAT_CONTEXT, WM_IO_WRITE, &ciscat->state, sizeof(ciscat->state)) < 0)
```
Comment (vikman90) [inline src/wazuh_modules/wm_ciscat.c]: This warning appears:
```
wazuh_modules/wm_ciscat.c: In function ‘wm_ciscat_main’:
wazuh_modules/wm_ciscat.c:109:25: warning: passing argument 1 of ‘wm_state_io’ from incompatible pointer type [-Wincompatible-pointer-types]
         if (wm_state_io(&WM_CISCAT_CONTEXT, WM_IO_WRITE, &ciscat->state, sizeof(ciscat->state)) < 0)
                         ^
In file included from wazuh_modules/wm_ciscat.c:14:0:
wazuh_modules/wmodules.h:107:5: note: expected ‘const char *’ but argument is of type ‘const wm_context * {aka const struct wm_context *}’
 int wm_state_io(const char * tag, int op, void *state, size_t size);
     ^~~~~~~~~~~
```

The correct function call is:
```c
wm_state_io(WM_CISCAT_CONTEXT.name, WM_IO_WRITE, &ciscat->state, sizeof(ciscat->state))
```

File: src/wazuh_modules/wm_ciscat.c [hunk 3]
```diff
@@ -0,0 +1,530 @@
...
+
+void wm_ciscat_setup(wm_ciscat *_ciscat) {
+    int i;
+
+    ciscat = _ciscat;
+    wm_ciscat_check();
+
+    // Read running state
+
+    if (wm_state_io(&WM_CISCAT_CONTEXT, WM_IO_READ, &ciscat->state, sizeof(ciscat->state)) < 0)
```
Comment (vikman90) [inline src/wazuh_modules/wm_ciscat.c]: This warning appears:
```
wazuh_modules/wm_ciscat.c: In function ‘wm_ciscat_setup’:
wazuh_modules/wm_ciscat.c:129:21: warning: passing argument 1 of ‘wm_state_io’ from incompatible pointer type [-Wincompatible-pointer-types]
     if (wm_state_io(&WM_CISCAT_CONTEXT, WM_IO_READ, &ciscat->state, sizeof(ciscat->state)) < 0)
                     ^
In file included from wazuh_modules/wm_ciscat.c:14:0:
wazuh_modules/wmodules.h:107:5: note: expected ‘const char *’ but argument is of type ‘const wm_context * {aka const struct wm_context *}’
 int wm_state_io(const char * tag, int op, void *state, size_t size);
     ^~~~~~~~~~~
```

The correct function call is:
```c
wm_state_io(WM_CISCAT_CONTEXT.name, WM_IO_READ, &ciscat->state, sizeof(ciscat->state))
```

File: src/wazuh_modules/wm_ciscat.c [hunk 4]
```diff
@@ -0,0 +1,530 @@
...
+            }
+
+            if (DeletePID("wazuh-modulesd:ciscat") < 0)
+                mterror_exit(WM_CISCAT_LOGTAG, "Couldn't delete PID file for child process: (%s)", strerror(errno));
+
+            _exit(0);
+
+        default:
+            // Parent process
+            switch(waitpid(-1, &child_status, 0)) {
```
Comment (vikman90) [inline src/wazuh_modules/wm_ciscat.c]: Are you sure you want to wait for **any** child process?
This implies that the wodle will continue when the script, or any child (like CIS-CAT or `xlstproc`) finishes.
I think that `waitpid(pid, &child_status, 0)` is better for you.

**Comment by chemamartinez** [comment]
Changes applied! Thanks @vikman90

**Pull request merged**